### ISSUE TITLE
ONNX Convert failed for types with scalar > 7

### ISSUE THREAD

**Issue by yoks** [pytorch/pytorch#32280]
## 🐛 Bug

https://github.com/pytorch/pytorch/blob/da7468853ae322252270bbb58032668bd21b7457/c10/core/ScalarType.h

Relevant line https://github.com/pytorch/pytorch/blob/da7468853ae322252270bbb58032668bd21b7457/c10/core/ScalarType.h#L15

It seems what 
https://github.com/pytorch/pytorch/blob/7df5dc2775e8baea4c64efafcc9efbece1e79ecc/torch/onnx/symbolic_helper.py#L508

Has ComplexHalf omitted, which in turn cause Index out of range exception for scalars greater > 7. For example for bool (11)

## To Reproduce

Try to convert model with bool types of tensor present, e.g. model which masks and uses fill placeholders like ones.

Produces following error:
```
File ".../torch/onnx/symbolic_opset9.py", line 1220, in ones
    value_t=torch.tensor([1], dtype=sym_help.scalar_type_to_pytorch_type[dtype]))
IndexError: list index out of range
```

## Expected behavior

No exception

## Environment

Please copy and paste the output from our
```
Collecting environment information...
PyTorch version: 1.3.1
Is debug build: No
CUDA used to build PyTorch: 10.1.243

OS: Ubuntu 19.10
GCC version: (Ubuntu 7.4.0-14ubuntu2) 7.4.0
CMake version: version 3.13.4

Python version: 3.7
Is CUDA available: Yes
CUDA runtime version: 10.2.89
GPU models and configuration: GPU 0: GeForce GTX 1050
Nvidia driver version: 440.36
cuDNN version: /usr/lib/x86_64-linux-gnu/libcudnn.so.7.6.5

Versions of relevant libraries:
[pip3] numpy==1.17.4
[pip3] pytorch-wpe==0.0.0
[pip3] torch==1.3.1
[pip3] torch-complex==0.0.1
[pip3] torchvision==0.4.2
[conda] blas                      1.0                         mkl  
[conda] mkl                       2019.4                      243  
[conda] mkl-service               2.3.0            py37he904b0f_0  
[conda] mkl_fft                   1.0.15           py37ha843d7b_0  
[conda] mkl_random                1.1.0            py37hd6b4f25_0  
[conda] pytorch                   1.0.1           py3.7_cuda10.0.130_cudnn7.4.2_2    pytorch
[conda] warpctc-pytorch10-cuda100 0.1.3                    pypi_0    pypi
[conda] warprnnt-pytorch          0.1                      pypi_0    pypi
```

## Additional context

cc @houseroad @spandantiwari @lara-hdr @BowenBao @neginraoof

**Comment by yoks** [comment]
Related issue
https://github.com/pytorch/pytorch/issues/25838

**Comment by yoks** [comment]
Well, technically it will not fail for types up to 10, but it will get wrong type from list because in accessed by index.

**Comment by garymm** [comment]
Fixed by #40006
